qiskit/passmanager/propertyset.py

```diff
@@ -10,15 +10,54 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""A property set is maintained by the pass runner.
+"""A property set dictionary that shared among optimization passes."""
 
 
-This is sort of shared memory space among passes.
-"""
+from dataclasses import dataclass, field
+from enum import IntEnum
+
+from .exceptions import PassManagerError
 
 
 class PropertySet(dict):
-    """A default dictionary-like object"""
+    """A default dictionary-like object."""
 
     def __missing__(self, key):
         return None
+
+
+class FencedPropertySet(PropertySet):
+    """A readonly property set that cannot be written via __setitem__."""
+
+    def __setitem__(self, key, value):
+        raise PassManagerError("The fenced PropertySet has the property __setitem__ protected.")
+
```

Comment:
I don't think so. This is used by flow controllers, but fenced object cannot protect mutable data anyway. I'm fine deprecation.

Reply:
If it's not too much trouble, I'd vote to deprecate and remove; it's barely used by us, and only really offers security theatre than actual access control.

Reply:
I removed the use of fenced property set in [e526bff](https://github.com/Qiskit/qiskit/pull/10127/commits/e526bffd1a365558fe84ba0a94b31d4e4b71d202) and DAGCircuit in [2051412](https://github.com/Qiskit/qiskit/pull/10127/commits/20514125c2870e30337e2d0d3a7f2556318a9184). I also needed to drop two unittests, because these objects are not protected anymore and some tests validate the protection.